### PR TITLE
install application dependencies during the image build

### LINKED ISSUE #332: Dockerfile for dashboard does not include yarn install
This makes a fresh clone of kubeapps not build successfully due to missing `react-scripts-ts` binary.

The following should be enough to run `yarn install` and only run it whenever included packages change - such as:

```
--- a/dashboard/Dockerfile
+++ b/dashboard/Dockerfile
@@ -1,5 +1,8 @@
 FROM node:8.9 AS build
 WORKDIR /app
+COPY package.json /app/package.json
+COPY yarn.lock /app/yarn.lock
+RUN yarn install
 COPY . /app
 RUN yarn run build
```

At this point `node_modules` could also be added to `dashboard/.dockerignore` so any local modules are not injected into the first stage of the build.

### PR DESCRIPTION
**Pull request by sameersbn**
fixes #332

### CODE REVIEW

File: dashboard/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,9 @@
 FROM node:8.9 AS build
 WORKDIR /app
+
+COPY package.json yarn.lock /app/
+RUN yarn install --frozen-lockfile
```
Comment (sameersbn) [inline dashboard/Dockerfile]: This improves docker cache hits in the development workflow

Comment (andresmgot) [reply dashboard/Dockerfile]: what do you mean with "this"? the `--frozen-lockfile`? Why not doing the `yarn install` after the `COPY . /app`?

Comment (sameersbn) [reply dashboard/Dockerfile]: > the --frozen-lockfile?

Apparently `yarn install` could go an update the yarn.lock file and update the package versions to something completely different from what it was tested against. For this reason the `--frozen-lockfile` is recommended for use in CI builds https://github.com/yarnpkg/yarn/issues/749

> Why not doing the yarn install after the COPY . /app?

Specifically w.r.t the developer use-case, this ensures that the docker layer cache gets reused during the builds. As a result image builds will be quicker when the dashboard sources are updated. Only when there are updates to the package.json/yarn.lock file, there will be a cache miss and `yarn install` will execute again.

Comment (andresmgot) [reply dashboard/Dockerfile]: ah, true

**Pull request merged**